### PR TITLE
feat: add FastifyPluginAsyncJsonSchemaToTs and FastifyPluginCallbackJsonSchemaToTs plugin types

### PR DESCRIPTION
**Pull request by Liam-Tait**
Add plugin specifc types

- FastifyPluginAsyncJStT
- FastifyPluginCallbackJStT

I opted to use an acronym for the type name instead of for bbrevity. If the full name is prefered, I am happy to update.

- FastifyPluginAsyncJSONSchemaToTs
- FastifyPluginCallbackJSONSchemaToTs

Based on the discussion around Typebox plugin exports [here](https://github.com/fastify/fastify/issues/3636), I thought it made sense for all the type providers to export something similar

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by Liam-Tait** [comment]
No script for npm run benchmark, ran npm run test all ok.

File: index.ts [hunk 1]
```diff
@@ -1,7 +1,47 @@
...
-import { JSONSchema7, FromSchema  } from 'json-schema-to-ts'
+import { JSONSchema7, FromSchema } from "json-schema-to-ts";
 
 export interface JsonSchemaToTsProvider extends FastifyTypeProvider {
-  output: this['input'] extends JSONSchema7 ? FromSchema<this['input']> : never
+  output: this["input"] extends JSONSchema7 ? FromSchema<this["input"]> : never;
 }
+
+/**
+ * FastifyPluginCallback with Typebox automatic type inference
```
Comment (RafaelGSS) [inline index.ts:17]: The references to Typebox should be updated

**Review by RafaelGSS — CHANGES_REQUESTED** [review]

File: index.ts (hunk 1 above)
Comment (Liam-Tait) [reply index.ts]: Fixed in 
[me, no I would never copy and paste code](https://github.com/fastify/fastify-type-provider-json-schema-to-ts/pull/15/commits/213cbf9bc9b6df6bad8fca67825b22e1336da98e)

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by Uzlopak** [comment]
@mcollina dont you think we should have the full name of the type and not an abbreviation?

**Review by RafaelGSS — APPROVED** [review]

**Comment by mcollina** [comment]
I'm mostly don't care about names of types as I don't code in TS. If things are problematic we'll get feedback about it.

**Comment by Uzlopak** [comment]
To be honest: I use typescript alot and I think he name of the type is bad. 

I would like to see them named longer. Maybe called them FastifyPluginAsyncInferTypeByJSONSchema or so.

**Comment by mcollina** [comment]
Then let's make it longer.

**Comment by Liam-Tait** [comment]
> I would like to see them named longer. Maybe called them FastifyPluginAsyncInferTypeByJSONSchema or so.

We would have to append `JSONSchemaToTs` instead of `JSONSchema` as this is JSONSchemaToTs specific and doesn't work with JSON schema alone / typebox / other jsonSchema helper functions.

I think `FastifyPluginAsyncInferTypeByJSONSchemaToTs` is overly verbose in real world usage. e.g.

```ts
const plugin: FastifyPluginAsyncInferTypeByJSONSchemaToTs = async (fastify, options) => {

}
```


I think there is value to have the naming of these types align with the typebox types and
future TypeProviders with `FastifyPluginAsync${Name}`

I' have updated this with the non abbreviated version for clarity.

```ts
const plugin: FastifyPluginAsyncJSONSchemaToTs = async (fastify, options) => {

}
```

File: README.md [hunk 2]
```diff
@@ -1,2 +1,33 @@
...
+
 A Type Provider for json-schema-to-ts
+
+## Plugin definition
+
+> **Note**
+> When using plugin types, withTypeProvider is not required in order to register the plugin
+
+```ts
+const plugin: FastifyPluginAsyncJStT = async function (fastify, _opts) {
```
Comment (Uzlopak) [inline README.md:11]:
```suggestion
const plugin: FastifyPluginAsyncJsonSchemaToTs = async function (fastify, _opts) {
```

Comment (Liam-Tait) [reply README.md:11]: Fixed with [remove all abbreviations](https://github.com/fastify/fastify-type-provider-json-schema-to-ts/pull/15/commits/7baf464e96b5271d1a97fd4a5bd83d31b16e4b6d)

**Comment by Liam-Tait** [comment]
@mcollina Types are updated removing the abbreviations. Believe ok to merge 👍

Cheers for the feedback everyone

**Review by climba03003 — APPROVED** [review]
LGTM.

**Pull request merged**